### ISSUE TITLE
abstract `this` modification does not respect `final`

### ISSUE THREAD

**Issue by nadako** [HaxeFoundation/haxe#9291]
I think this should fail with the `Cannot assign to final` error (or something more explanatory, since we're not explicitly assigning here).

```haxe
abstract A(Int) from Int {
	public inline function set(value:Int) {
		this = value;
	}
}

class Main {
	static function main() {
		final a:A = 1;
		a.set(2);
	}
}
```

**Comment by RealyUniqueName** [comment]
I'm not sure about this.
Re-assigning `this` is an implementation detail of an abstract. And on usage site things _looks_ correct.
I guess it boils down to how Haxe specify `final`. 
Does it imply that a `final` value stored into a temporary variable will always hold `==` with the original `final` value?

**Comment by terurou** [comment]
Is it already implemented in Haxe 4.3.0?

This code is working in 4.2.5, but not working in 4,3.0.
https://try.haxe.org/#2A20ed29

**Comment by kLabz** [comment]
Ah yeah, this can indeed be closed now. Thanks!
